packages/manager/src/managers/telemetry/TelemetryManager.ts

```diff
@@ -0,0 +1,142 @@
+import { randomUUID } from "node:crypto";
+
+import SegmentClient from "analytics-node";
+
+import { readPrismicrc } from "../../lib/prismicrc";
+
+import { API_TOKENS } from "../../constants/API_TOKENS";
+
+import { BaseManager } from "../BaseManager";
+
+import {
+	HumanSegmentEventType,
+	HumanSegmentEventTypes,
+	SegmentEvents,
+} from "./types";
+
+type TelemetryManagerTrackArgs = SegmentEvents;
+
+type TelemetryManagerIdentifyArgs = {
+	userID: string;
+	intercomHash: string;
+};
+
+function assertTelemetryInitialized(
+	segmentClient: SegmentClient | undefined,
+): asserts segmentClient is NonNullable<typeof segmentClient> {
+	if (segmentClient == undefined) {
+		throw new Error(
+			"Telemetry has not been initialized. Run `SliceMachineManager.telemetry.prototype.initTelemetry()` before re-calling this method.",
+		);
+	}
+}
+
+export class TelemetryManager extends BaseManager {
+	private _enabled = false;
+	private _segmentClient: SegmentClient | undefined = undefined;
+	private _anonymousID: string | undefined = undefined;
+	private _userID: string | undefined = undefined;
+
+	async initTelemetry(): Promise<void> {
+		if (this._segmentClient) {
+			// Prevent subsequent initializations.
+			return;
+		}
+
+		let root: string;
+		try {
+			root = await this.project.getRoot();
+		} catch {
+			root = await this.project.suggestRoot();
+		}
+
+		this._enabled = readPrismicrc(root).telemetry !== false;
+		this._segmentClient = new SegmentClient(API_TOKENS.SegmentKey, {
+			// Since it's a local app, we do not benefit from event batching the way a server would normally do, all tracking event will be awaited.
+			flushAt: 1,
+			// TODO: Verify that this actually does not send data to Segment when false.
+			enable: this._enabled,
+		});
+		this._anonymousID = randomUUID();
+	}
+
+	// TODO: Should `userId` be automatically populated by the logged in
```

Comment:
Okay so I took a look at how authenticating and the sessions are working on the main branch.

In short, the getEnv will take the cookie from `~/.prismic` (if it's there) one then use that cookie to fetch the UsersProfile which contains their shortId, every call to the tracking route initially calls `getEnv` and checks for the `shortId` If there's not one then an anonymous id is used, and eventually the two are related when the user logs in.

Important details is the _shortId_ and not the _userId_.

To answer your question, i depends how long these instances are active for. Like if in the server it's initialised on each request, it's nice to have but not essential since the class will be initialised on each request. if that's the case, I'm not 100% sure it is yet. (Although there would be a performance improvement if it wasn't).

For the init I think (not looked at it yet) but I would imagine the instance is more longer lived... so yes :)

Reply:
The `@slicemachine/manager` server is long-living; each manager class and its state exists for the entire lifetime of the process.

We wouldn't cache reading `~/.prismic`, so we would always have the most up-to-date `shortId`/`userId` anytime we call `track().

Based on that info, it sounds like we could automatically populate the Segment event's `userId` property with the value from `~/.prismic` without needing to pass it to each `track()` call.

Does that sound right to you?

I don't totally understand the difference between `shortId` and `userId`. Is `shortId` only for Intercom? Which one should we be sending to Segment?

Reply:
userId is somthing we using internally, while shortId is wha we send to other services. I think intercom uses a hash.

it's a bit confusing he we have a userId from prismic, and another for sending to segment (shortID). 

https://github.com/prismicio/slice-machine/blob/master/packages/slice-machine/server/src/api/tracking.ts#L51-L56

I think @Heyrows knows a bit more about what data segment and intercom/amplitude need :)